include/zenoh-pico/config.h

```diff
@@ -232,13 +232,6 @@
 #define Z_FEATURE_FRAGMENTATION 1
 #endif
 
-/**
- * Enable attachments.
- */
-#ifndef Z_FEATURE_ATTACHMENT
```

Comment:
With the change, the impact on code size should be minimal since all the code is now on the API, and it shouldn't be linked if not used by the application.

So I believe we should remove this token so users have one less config option to worry about